analyzers/src/SonarAnalyzer.CSharp/Rules/TypeMemberVisibility.cs

```diff
@@ -37,59 +38,70 @@ public sealed class TypeMemberVisibility : SonarDiagnosticAnalyzer
         private const string MessageFormat = "Types should not have members with visibility set higher than the type's visibility";
 
         private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        private static readonly SyntaxKind[] TypeKinds = { SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration, SyntaxKind.EnumDeclaration, SyntaxKindEx.RecordClassDeclaration };
+        private static readonly SyntaxKind[] TypeKinds =
+        {
+            SyntaxKind.ClassDeclaration,
+            SyntaxKind.EnumDeclaration,
+            SyntaxKindEx.RecordClassDeclaration,
+            SyntaxKindEx.RecordStructDeclaration,
+            SyntaxKind.StructDeclaration,
+        };
         private static readonly SyntaxKind[] MemberDeclarationKinds =
         {
             SyntaxKind.ClassDeclaration,
             SyntaxKind.ConstructorDeclaration,
             SyntaxKind.DelegateDeclaration,
+            SyntaxKind.EnumDeclaration,
```

Comment:
Let's just remove this whole field. It's used here:
```
                           .Where(node => node.IsAnyKind(MemberDeclarationKinds))
                           .OfType<MemberDeclarationSyntax>()
```
to filter something, that is by-design desired to be `MemberDeclarationSyntax`. So we can do just this instead, without a risk of missing something.
```
                           .OfType<MemberDeclarationSyntax>()
```

Reply:
I did so. I just needed to exclude operators. these are blacklisted instead of whitelisted now.

Reply:
Waaay better! It's now obvious what it actually does